basic_pitch/data/tf_example_deserialization.py

```diff
@@ -0,0 +1,697 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright 2024 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import uuid
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple
+
+import numpy as np
+import tensorflow as tf
+
+# import tensorflow_addons as tfa
+
+from basic_pitch.constants import (
+    ANNOTATIONS_FPS,
+    ANNOT_N_FRAMES,
+    AUDIO_N_CHANNELS,
+    AUDIO_N_SAMPLES,
+    AUDIO_SAMPLE_RATE,
+    AUDIO_WINDOW_LENGTH,
+    N_FREQ_BINS_NOTES,
+    N_FREQ_BINS_CONTOURS,
+)
+
+N_SAMPLES_PER_TRACK = 20
+
+
+def prepare_datasets(
+    datasets_base_path: str,
+    training_shuffle_buffer_size: int,
+    batch_size: int,
+    validation_steps: int,
+    datasets_to_use: List[str],
+    dataset_sampling_frequency: np.ndarray,
+) -> Tuple[tf.data.Dataset, tf.data.Dataset]:
+    """
+    Return a training and a testing dataset.
+
+    Args:
+        training_shuffle_buffer_size : size of shuffle buffer (only for training set)
+        batch_size : ..
+    """
+    assert batch_size > 0
+    assert validation_steps is not None and validation_steps > 0
+    assert training_shuffle_buffer_size is not None
+
+    # init both
+    ds_train = sample_datasets(
+        "train",
+        datasets_base_path,
+        datasets=datasets_to_use,
+        dataset_sampling_frequency=dataset_sampling_frequency,
+    )
+    ds_validation = sample_datasets(
+        "validation",
+        datasets_base_path,
+        datasets=datasets_to_use,
+        dataset_sampling_frequency=dataset_sampling_frequency,
+    )
+
+    # check that the base dataset returned by ds_function is FINITE
+    for ds in [ds_train, ds_validation]:
+        tf.debugging.assert_none_equal(
```

Comment:
I think this can be a regular assert. It doesn't have to be a tf.debugging.assert

Reply:
this seems like it could be useful though? would an assert != be the same as an element wise check? would you prefer a loop of asserts instead?